pymc3/tests/test_distributions_random.py

```diff
@@ -97,6 +97,15 @@ def test_random_sample_returns_nd_array(self):
         assert isinstance(mu, np.ndarray)
         assert isinstance(tau, np.ndarray)
 
+    def test_random_sample_returns_correctly(self):
+        # Based on what we discovered in #GH2909
+        with pm.Model():
+            a = pm.Uniform('a', lower=0, upper=1, shape=10)
+            b = pm.Binomial('b', n=1, p=a, shape=10)
+        array_of_randoms = b.random(size=10000).mean(axis=0)
+        npt.assert_allclose(array_of_randoms, [0.5338, 0.5443, 0.5313, 0.5392, 0.5372, 0.5435, 0.5287, 0.5432,
```

Comment:
heh, now this is surprising to me that these values are all above 0.5.  Any guess why?

Reply:
I think it varies a bit by the run. That's just one run of it. Is this wrong?

Reply:
It is random so anything can happen! I am expecting this test case to be equivalent to

```
samples = []
for _ in range(10000):
    a = np.random.uniform(0, 1, 10)
    samples.append(np.random.binomial(n=1, p=a))
np.array(samples).mean(axis=0)
```
and it is very rare to see any values in the resulting array be outside of (0.48, 0.52), much less all 10 values (and less again all 10 being greater than 0.5). 

I would guess there is still a subtle bug, or else I am thinking of this wrong.